### PR TITLE
Tc/add services

### PR DESCRIPTION
**Pull request by CedThen**

### CODE REVIEW

**Comment by CedThen** [comment]
O i forgot to mention, order is hidden and will default to being added to the end. I think a drag/drop to reorder services is pbly more intuitive? 

The previous pull request comes first btw

File: src/components/Main/Schedule/Home.tsx [hunk 1]
```diff
@@ -1,51 +1,107 @@
...
   );
+  const [mutateAddSchedule] = useMutation(addSchedule, {
+    onSuccess: () => cache.invalidateQueries('schedulesData'), //causes the schedulesData query to call and update on success
+  });
+  const [mutateAddService] = useMutation(addService, {
+    onSuccess: () => cache.invalidateQueries('schedulesData'),
+  });
 
-  if (status === 'loading') return <div>loading...</div>; // loading state
+  // if (status === 'loading') return <div>loading...</div>; // replaced by useSpinner() ??
```
Comment (hchiam) [inline src/components/Main/Schedule/Home.tsx]: yeah prolly can just delete this line. i remember trying to get this loading div to show up, but it doesn't show up on my machine.

File: src/components/Main/Schedule/Home.tsx [hunk 2]
```diff
@@ -1,51 +1,107 @@
...
 import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
 import { buttonTheme } from '../../../shared/styles/theme.js';
 import { useSpinner } from '../../../shared/styles/loading-spinner';
 
+import AddIcon from '@material-ui/icons/Add';
+
 export const Home = () => {
   const classes = useStyles();
-  useSpinner();
+  // useSpinner();
```
Comment (hchiam) [inline src/components/Main/Schedule/Home.tsx:24]: again, I tried this branch too (`tc/add-services`), and the spinner seems to work fine on my machine, so you could uncomment this. 

But if you find that the spinner is flickering or stuck for you, then I'll look into giving you a helper function, maybe something like `loadingSpinnerOn(false)`, which you could call upon API call completion.

File: src/components/Main/Schedule/Home.tsx [hunk 3]
```diff
@@ -57,13 +113,26 @@ export const Home = () => {
...
       />
       <div className={classes.schedulesContainer}>
-        {schedules?.map((schedule: any, idx: any) => (
+        <button onClick={onAddServiceClick}>
+          <AddIcon height={50} width={50} /> Add New Service
+        </button>
+        {displayedSchedule?.map((schedule: any, idx: any) => (
           <Scheduler schedule={schedule} key={idx} />
         ))}
       </div>
```
Comment (hchiam) [inline src/components/Main/Schedule/Home.tsx:136]: I think in the mocks this is under the bottom table, but I can change this when I do the styling later.

File: src/components/Main/Schedule/ScheduleTabs.tsx [hunk 4]
```diff
@@ -14,15 +14,15 @@ import {
 import { makeStyles, Theme, createStyles } from '@material-ui/core/styles';
 import { useSpinner } from '../../../shared/styles/loading-spinner';
 
-export const ScheduleTabs = ({ tabIdx, handleChange, titles }: ScheduleTabsProps) => {
+export const ScheduleTabs = ({ tabIdx, onTabClick, titles }: ScheduleTabsProps) => {
   const classes = useStyles();
-  useSpinner();
+  // useSpinner();
```
Comment (hchiam) [inline src/components/Main/Schedule/ScheduleTabs.tsx:19]: see other comment

File: src/components/shared/FormDialog.tsx [hunk 5]
```diff
@@ -61,6 +61,7 @@ export function FormDialog({ handleClose, isOpen, title }: AddUserProps) {
 
   // validations needed
   // valid email provided
+  // why are the submit/cancel buttons listitems?
```
Comment (hchiam) [inline src/components/shared/FormDialog.tsx:64]: 🤷 Maybe whoever wrote this originally was trying to make them go on separate lines.

**Review by hchiam — CHANGES_REQUESTED** [review]

File: src/components/Main/Schedule/NewScheduleForm.tsx [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+  }
+
+  function onSubmitForm() {
+    setErrorMessage('');
+    let msg: string = '';
+    if (scheduleTitle.length === 0 || scheduleTitle.length >= 32)
+      msg = '-Title must be not be blank and be under 32 characters long'; //line breaks don't work ahhh
+    if (new Date(endDate) <= new Date(startDate)) msg += '\n-Invalid Date Range';
+    if (team === 0) msg += '\n-Please assign a team to this schedule';
+    if (msg.length > 0) return setErrorMessage(msg);
```
Comment (hchiam) [inline src/components/Main/Schedule/NewScheduleForm.tsx]: Having trouble making line breaks work? Does using `<br/>` work?
```suggestion
    if (scheduleTitle.length === 0 || scheduleTitle.length >= 32)
      msg = '-Title must be not be blank and be under 32 characters long'; //line breaks don't work ahhh
    if (new Date(endDate) <= new Date(startDate)) msg += '<br/>-Invalid Date Range';
    if (team === 0) msg += '<br/>-Please assign a team to this schedule';
    if (msg.length > 0) return setErrorMessage(msg);
```

File: src/components/Main/Schedule/NewScheduleForm.tsx [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+  const classes = useStyles();
+
+  //needed to format date so that the date picker can display it properly
+  function toDateString(date: Date): string {
+    return `${date.getFullYear()}-${date.getMonth()}-${date.getDate()}`;
+  }
+
+  function NewlineText(props: { text: string }) {
+    const text = props.text;
+    const newText = text.split('\n').map((str) => <p>{str}</p>);
```
Comment (hchiam) [inline src/components/Main/Schedule/NewScheduleForm.tsx]: If `\n` isn't working maybe just do this?
```suggestion
    const newText = text.replaceAll('\n', '<br/>');
```

**Review by hchiam — CHANGES_REQUESTED** [review]

File: src/components/Main/Schedule/Home.tsx (hunk 2 above)
Comment (hchiam) [reply src/components/Main/Schedule/Home.tsx:24]: @CedThen - I got a fix for the spinner [here](https://github.com/House-on-a-Rock/tjc-scheduling-web/pull/36/files#diff-7b4f7e8958ec472ac00c2ace4e37d1e0f6b4508eb484c2d006e73be0e52a3695)

File: src/components/Main/Schedule/Home.tsx (hunk 3 above)
Comment (CedThen) [reply src/components/Main/Schedule/Home.tsx:136]: ok thanks

File: src/components/Main/Schedule/NewScheduleForm.tsx (hunk 7 above)
Comment (CedThen) [reply src/components/Main/Schedule/NewScheduleForm.tsx]: I got a new way of handling validation and form input errors, but i'll try to remember this for next time. I guess the line breaks don't work inside jsx or something

**Pull request merged**